services/venice-controller/src/main/java/com/linkedin/venice/controller/StoreDeletionValidationUtils.java

```diff
@@ -0,0 +1,329 @@
+
+import com.linkedin.venice.common.VeniceSystemStoreType;
+import com.linkedin.venice.helix.ZkStoreConfigAccessor;
+import com.linkedin.venice.meta.Store;
+import com.linkedin.venice.meta.StoreConfig;
+import com.linkedin.venice.pubsub.api.PubSubTopic;
+import com.linkedin.venice.pubsub.manager.TopicManager;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * Utility class for Venice store deletion validation operations.
+ * Contains methods to validate store deletion and other store-related validation logic.
+ * 
+ * This class provides a clean separation of store validation logic from the main
+ * VeniceHelixAdmin class, making it easier to test and maintain.
+ */
+public class StoreDeletionValidationUtils {
+  private static final Logger LOGGER = LogManager.getLogger(StoreDeletionValidationUtils.class);
+
+  private StoreDeletionValidationUtils() {
+    // Utility class - prevent instantiation
+  }
+
+  /**
+   * Validates that a store has been completely deleted from the Venice cluster.
+   * This method performs comprehensive checks across multiple subsystems to ensure
+   * no lingering resources remain that would prevent safe store recreation.
+   *
+   * Resources checked:
+   * 1. Store configuration in ZooKeeper
+   * 2. Store metadata in store repository
+   * 3. System stores (META_STORE and DAVINCI_PUSH_STATUS_STORE)
+   * 4. Kafka topics (version, RT, and system store topics)
+   * 5. Helix resources
+   *
+   * @param admin the Venice admin instance for performing validation operations
+   * @param clusterName the name of the cluster to check
+   * @param storeName the name of the store to validate deletion for
+   * @return StoreDeletedValidation indicating whether the store is fully deleted or what resources remain
+   */
+  public static StoreDeletedValidation validateStoreDeleted(Admin admin, String clusterName, String storeName) {
+    final StoreDeletedValidation result = new StoreDeletedValidation(clusterName, storeName);
+
+    try {
+      // 1. Check if Store Config still exists in ZooKeeper
+      if (checkStoreConfig(admin, result, clusterName, storeName)) {
+        return result;
+      }
+
+      // 2. Check if Store metadata still exists in store repository
+      if (checkStoreMetadata(admin, result, clusterName, storeName)) {
+        return result;
+      }
+
+      // 3. Check system store metadata
+      if (checkSystemStoreMetadata(admin, result, clusterName, storeName)) {
+        return result;
+      }
+
+      // 4. Check Kafka topics: version topics, RT topics, and system store topics
+      if (checkKafkaTopics(admin, result, clusterName, storeName)) {
+        return result;
+      }
+
+      // 5. Check Helix resources for main store
+      if (checkMainStoreHelixResources(admin, result, clusterName, storeName)) {
```

Comment:
Can we move the Helix resources check before the topic resources check? This is because based on the order we perform cleanup (delete metadata -> delete helix resource -> delete topic) if helix resources still exists then corresponding topic definitely also exists.